guardian/typerighter
# Batch edit rules (client-side)

## What does this change?
Enable batch edit on the client side to accompany server-side implementation (#346). This PR introduces a new rule form `RuleFormBatchEdit` and a new hook `useBatchRules` to allow users to edit multiple rules at the same time. Note that I've had to make changes to `DbRuleDraft.batchUpdate` to take into consideration requests where only category or tags are edited, not both.

With the current implementation, upon selecting multiple rules, the form will display the category of the first selected rule, and the tags section will show all the tags of the selected rules.
![batch-edit](https://github.com/guardian/typerighter/assets/47318984/bd9e9d6b-eccb-4be5-8c93-782342fc8d38)



## How to test

1. Select multiple rules for testing using the tick boxes.
2. Update the category and/or the tags section and hit the update button. In the dev console, check that a POST request is made to `/rules/batch` endpoint with the updated fields as the payload. If either of category of tags was edited, the request should only include the relevant fields.
3. The new rules fetched in the table should now be updated with the new tags and/or categories.
